broadway_api/daos/base.py

```diff
@@ -1,15 +1,44 @@
+import bson
+import logging
+
+from pymongo import MongoClient
+from pymongo.collection import Collection
+
+logger = logging.getLogger("dao")
+
+
+def validate_obj_size(func):
+    def wrapper(*args, **kwargs):
+        base_dao: BaseDao = args[0]
+        obj = args[1]
+
+        if base_dao.is_obj_size_valid(obj):
+            return func(*args, **kwargs)
+
+        logger.critical(
+            "Bson document larger than the maximum bson size as specified by the mongo"
+            "client. Not saving the object."
+        )
+        return
+
+    return wrapper
+
+
 class BaseDao:
     ID = "_id"
 
     def __init__(self, app):
-        self._config = app["CONFIG"]
-        self._client = app["DB"]
+        self._config: dict = app["CONFIG"]
+        self._client: MongoClient = app["DB"]
 
-    def _get_collection(self, db_name, collection_name):
+    def _get_collection(self, db_name, collection_name) -> Collection:
         return self._client[db_name][collection_name]
 
-    def _to_store(self, obj):
+    def _to_store(self, obj) -> dict:
         raise NotImplementedError("_to_store not implemented")
 
-    def _from_store(self, obj):
+    def _from_store(self, obj) -> object:
```

Comment:
Nit: technically the return type is `Optional["BaseDao"]` (note the self-reference).